src/NServiceBus.AcceptanceTests/Routing/SubscriptionBehavior.cs

```diff
@@ -36,16 +44,18 @@ public SubscriptionBehavior(Action<SubscriptionEventArgs, TContext> action, TCon
 
         static string GetSubscriptionMessageTypeFrom(IncomingMessage msg)
         {
-            return (from header in msg.Headers where header.Key == Headers.SubscriptionMessageType select header.Value).FirstOrDefault();
+            string headerValue;
+            return msg.Headers.TryGetValue(Headers.SubscriptionMessageType, out headerValue) ? headerValue : null;
         }
 
         Action<SubscriptionEventArgs, TContext> action;
         TContext scenarioContext;
+        MessageIntentEnum intentToHandle;
 
         internal class Registration : RegisterStep
         {
-            public Registration()
-                : base("SubscriptionBehavior", typeof(SubscriptionBehavior<TContext>), "So we can get subscription events")
+            public Registration(string id, Func<IBuilder, IBehavior> behaviorFactory)
+                : base(id, typeof(SubscriptionBehavior<TContext>), "notify subscription events", behaviorFactory)
             {
                 InsertBeforeIfExists("ProcessSubscriptionRequests");
```

Comment:
Wonder if w can ditch this and the entire registration since we only call `OnEndpointUnsubscribed|Subscribed` in message driven pubsub tests? (ie the  ProcessSubscriptionRequests behavior should always be present)